src/autoscaler/scalingengine/server/applock_test.go

```diff
@@ -0,0 +1,106 @@
+package server_test
+
+import (
+	. "autoscaler/scalingengine/server"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"time"
+)
+
+var (
+	ch      chan string
+	appLock *AppLock
+	err     interface{}
+)
+
+var _ = Describe("Applock", func() {
+	BeforeEach(func() {
+		appLock = NewAppLock()
```

Comment:
@fraenkel 
there is a data race between line 20(write appLock) and line 30/95 (read appLock).  Theoretically there is guarantee that line 20 is executed before line 30 go routine is created and line 95 is executed.  Seems like it is because they are too close?  Is there good way to handle it without introducing weird code?